src/views/Administration/AddEstablishment.vue

```diff
@@ -0,0 +1,281 @@
+      modal-id="SearchModal" title="Establecimiento no encontrado"
+      btn-accept="Nuevo" btn-reject="Inicio"
+      @acceptModal="this.$router.push('/establecimiento/formulario')"
+      @rejectModal="this.$router.push('/')"
+  ></confirmation-modal>
+</template>
+<script>
+import {mapActions, mapState} from "vuex";
+import ConfirmationModal from "../../components/ConfirmationModal";
+import { Modal } from 'bootstrap';
+
+
+export default {
+  name: "AddEstablishment",
+  data() {
+    return {
+      success: null,
+      edit: false,
+      owners:[],
+      errorSave: {
+        name: false,
+        cuig: false,
+        owner: false
+      },
+
+    };
+  },
+  components:{
+    ConfirmationModal
+  },
+  mounted() {
+    this.searchOwners()
+    this.cuig = this.$route.query.cuig;
+    console.log(this.cuig);
+    this.edit = !(this.cuig == null || this.cuig === "");
+    console.log(this.edit)
+    if (this.edit){
+      this.formRetrieveEstablishment()
+    }else{
+      this.clearEstablishmentData();
+    }
+  },
+  computed: {
+    ...mapState("establishment", ["establishment", "error","owner"]),
+    
+  },
+  methods: {
+    ...mapActions("establishment", ["getEstablishment", "saveEstablishment", "deleteEstablishment", "clearEstablishmentData","listOwners"]),
+
+    successCall(message) {
+      this.success = message;
+      this.edit = false;
+      this.clearEstablishmentData();
+    },
+    async formSaveEstablishment() {
+      this.showModal = true
+      this.errorSave = {};
+      this.success = null;
+      this.errorSave = {
+        name: (this.establishment.name == null || this.establishment.name === ""),
+        cuig: (this.establishment.cuig == null || this.establishment.ciug === ""),
+        owner: (this.establishment.owner == null || this.establishment.owner === "")
+      }
+      if (this.errorSave.name || this.errorSave.cuig || this.errorSave.owner) {
+        console.error(this.errorSave)
+        return
+      }
+      let data = {
+        edit: this.edit,
+        establishment: this.establishment
+      };
+      this.saveEstablishment(data).then(
+        () => {
+          this.successCall("El Establecimiento se guardó correctamente");
+        }
+      );
+    },
+    async formRetrieveEstablishment() {
+      this.errorSave = {};
+      this.success = null;
+      this.getEstablishment(this.cuig).then(
+        () => {},
+        (error) => {
+          let modal = new Modal(document.getElementById('SearchModal'));
+          modal.show()
+          this.edit = false;
+        }
+      )
+    },
+   
+    async formDeleteEstablishment() {
+      this.errorSave = {};
+      this.success = null;
+      this.deleteEstablishment(this.cuig).then(
+        () => {
+          this.successCall("El Establecimiento se elimino correctamente");
+        }
+      );
+    },
+    async searchOwners() {
+      this.owners=null;
+      this.listOwners('patron').then(
```

Comment:
deberiamos dejar la aclaracion de que buscamos patrones en los service

Reply:
Te referis a un comentario?

Reply:
no no, pasarlo a los services, y no que sea una peticion desde la pagina

Reply:
Pero no seria mejor asi , para que quede reutilizable por si lo necesitamos con otro tipo de usuario

Reply:
basicamente en esa linea quedaria `this.listOwners().then(` y la funcion listOwners del store se encargaria de pasar el parametro, ya que es lo que uno esperaria de listOwners

Reply:
@agustinsuso

Reply:
Corregido